### PR TITLE
Add Save Settings to File Button

### LINKED ISSUE #28606: Save Settings to File
**Original reporter:** SANS group

It would be nice to be able to save the default settings to a file for sharing with others. This could be done for All settings or just for Plot Settings, which would help getting the defaults set nicely on many beamline computers in a group!

### PR DESCRIPTION
**Pull request by alicerussell1**
**Description of work.**
This PR adds a button to the settings to save the settings to a file and another to load the settings back in.

**Report to:** SANS group

**To test:**

Open workbench and open settings (`File`->`Settings`)
Change some settings and click `Save to File`. This should create a file which contains all the settings in their current state.
Change some more settings
Click `Load from File` and load in the saved file.
This should put all the settings back into the state they were when they were saved.

Fixes #28606 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/applications/workbench/workbench/widgets/settings/model.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                    if CONF.has(prop) or prop.find('/') != -1:
+                        if prop == GeneralProperties.USER_LAYOUT.value:
+                            pass
+                        elif prop == GeneralProperties.FONT.value and not setting[1]:
+                            # if the font setting is empty removing it will make workbench use the default
+                            CONF.remove(prop)
+                        else:
+                            try:
+                                value = ast.literal_eval(setting[1])
+                                pass
```
Comment (ConorMFinn) [inline qt/applications/workbench/workbench/widgets/settings/model.py:45]: Is this `pass` neccessary here?

**Review by cailafinn — CHANGES_REQUESTED** [review]
Just one very minor thing I found in the code, otherwise looks good. Just a couple of other more subjective notes:
- On ubuntu at least, the buttons appear squashed together. Could some space be added between the save, load, and close buttons?
- Once the settings have been saved or loaded, the view returns to the main workbench window but the settings are still open but hidden and nothing is clickable. This may be something to do with the modal windows issue we've had in the past. Is there a way to reliably bring the settings window back to the front after the file dialog is closed?

File: qt/applications/workbench/workbench/widgets/settings/model.py (hunk 1 above)
Comment (alicerussell1) [reply qt/applications/workbench/workbench/widgets/settings/model.py:45]: Ah no it isn't, I meant to remove it - good catch!

**Review by cailafinn — DISMISSED** [review]
Code looks good and functions as described. Nice work 👍

**Review by cailafinn — DISMISSED** [review]

**Comment by alicerussell1** [comment]
@ConorMFinn There was another conflict from adding the default colormap setting, should be fixed now

**Review by cailafinn — APPROVED** [review]

**Comment by gemmaguest** [comment]
@alicerussell1 It was requested to report this to the SANS group when merged

**Pull request merged**